src/twisted/test/test_defer.py

```diff
@@ -2943,3 +2943,24 @@ def test_willNotAllowNonDeferredOrCoroutine(self):
         """
         with self.assertRaises(ValueError):
             defer.ensureDeferred("something")
+
+
+
+class TimeoutErrorTests(unittest.TestCase, ImmediateFailureMixin):
+    """
+    L{twisted.internet.defer} timeout code.
+    """
+    def test_deprecatedTimeout(self):
+        """
+        L{twisted.internet.defer.timeout} is deprecated.
+        """
+        deferred = defer.Deferred()
+        defer.timeout(deferred)
+        self.assertFailure(deferred, defer.TimeoutError)
+        warningsShown = self.flushWarnings([self.test_deprecatedTimeout])
+        self.assertEqual(len(warningsShown), 1)
+        self.assertIdentical(warningsShown[0]['category'], DeprecationWarning)
```

Comment:
I think `assertIs` is preferred.  That's the stdlib unittest spelling for this assertion.